providers/implementations/kdfs/hkdf.c

```diff
@@ -180,6 +186,37 @@ static size_t kdf_hkdf_size(KDF_HKDF *ctx)
     return sz;
 }
 
+#ifdef FIPS_MODULE
+static size_t fips_key_size(KDF_HKDF *ctx)
+{
+    const EVP_MD *md = ossl_prov_digest_md(&ctx->digest);
+    size_t key_size = 0;
+
+    if (ctx->key != NULL)
+        key_size = ctx->key_len;
+    else if (md != NULL)
```

Comment:
I dont understand what this is doing..
kdf_hkdf_derive() checks that both ctx->key and md are not NULL..

Reply:
For the MD case you would need to point to some evidence that this is the correct thing to do.. (For signatures at least the security strength is half of the bitsize - not sure it applies to KDF's)

Reply:
This should be only for the TLS1.3 case. The code updated.

According to
* RFC 8446
  ```
  If a given secret is not available, then the 0-value consisting of a
  string of Hash.length bytes set to zeros is used.
  ```
* SP 800-131Ar2
  ```
  The use of HMAC-based KDFs is acceptable using a hash function specified in FIPS 180
  or FIPS 202 with a key whose length is at least 112 bits
  ```

Therefore, if key is not available, I just get the MD length as the key size.